precompiles/outposts/osmosis/types_test.go

```diff
@@ -6,15 +6,17 @@ import (
 	"testing"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
 
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
 	osmosisoutpost "github.com/evmos/evmos/v15/precompiles/outposts/osmosis"
 	"github.com/evmos/evmos/v15/utils"
-	"github.com/stretchr/testify/require"
 )
 
 func TestCreatePacketWithMemo(t *testing.T) {
```

Comment:
Please, assert in this test the exact expected memo. I understand should have the following structure:

```
{"wasm": {
    "contract": "osmo1crosschainswapscontract", 
    "msg": {
        "osmosis_swap": {
            "output_denom":"token1",
            "slippage":{"twap": {"slippage_percentage":"20", "window_seconds": 10}},
            "receiver":"juno1receiver",
            "on_failed_delivery": "do_nothing",
            "next_memo":null
        }
    }
}}
```

With the current implementation, we're getting this structure:

```
{
  "memo": {
    "wasm": {
      "contract": "evmos1vl0x3xr0zwgrllhdzxxlkal7txnnk56q3552x7",
      "msg": {
        "osmosis_swap": {
          "output_denom": "aevmos",
          "slippage": {
            "twap": {
              "slippage_percentage": 10,
              "window_seconds": 30
            }
          },
          "receiver": "evmos1vl0x3xr0zwgrllhdzxxlkal7txnnk56q3552x7",
          "on_failed_delivery": "do_nothing",
          "next_memo": "a next memo"
        }
      }
    }
  }
}
```

notice that there's an extra level with the `memo` key. Maybe we can remove this

Reply:
Good catch!